addons/resource/models/resource_mixin.py

```diff
@@ -153,8 +197,8 @@ def list_leaves(self, from_datetime, to_datetime, calendar=None, domain=None):
         if not to_datetime.tzinfo:
             to_datetime = to_datetime.replace(tzinfo=utc)
 
-        attendances = calendar._attendance_intervals(from_datetime, to_datetime, resource)
-        leaves = calendar._leave_intervals(from_datetime, to_datetime, resource, domain)
+        attendances = calendar._attendance_intervals_batch(from_datetime, to_datetime, resource)[resource.id]
+        leaves = calendar._leave_intervals_batch(from_datetime, to_datetime, resource, domain)[resource.id]
```

Comment:
This API change introduces a lot of noise without any benefit in those non-batched cases.  It is even worse, from a readability point of view.  I suggest to keep the old API for the non-batched cases, as this change is simply useless and risky in this pull request.

Imagine we have to reinvent this API on master.  How would we do it?  I probably would go for two different APIs:
```python
# for no resource: return a single element
item = calendar._attendance_intervals(from_datetime, to_datetime)

# for resources: map each resource to its result in a list
items = calendar._resource_attendance_intervals(from_datetime, to_datetime, resources)
```
Can we make this branch already *prepare* the ground for a better API in the future?  This could avoid changing it again in master.

Reply:
I agree for the solution "1 no resource / 1 batch method". 

However, this implies a signature change, since ``_attendance_intervals`` would be related to the no-resource case, and since the current signature handles a optional ``resource`` parameter.

This will still lead to some notations like
```python3
leaves = calendar._resources_leave_intervals(from_datetime, to_datetime, resource, domain)[resource.id]
```

So I imagine that those changes should be handled in master only to get the result you propose.

Reply:
Well, we *can* leave the non-batched version *as is*, and possibly deprecate the support of `resource` in master, and then remove it in a future version.  All we need for now is *add* the batched version, and *encourage* people to use it instead of the other one.